talipp/indicators/VWAP.py

```diff
@@ -0,0 +1,94 @@
+# =============================================================================
+# Created By  : Pramit Biswas
+# Created Date: Sun Jun 06 19:55:00 IST 2021
+# =============================================================================
+
+from typing import List, Any
+
+from talipp.indicators.Indicator import Indicator
+from talipp.ohlcv import OHLCV
+
+
+class VWAP(Indicator):
+    """
+    Volume Weighted Average Price
+    Output: a list of floats
+    ALERT: This does not check season starting.
+    """
+
+    def __init__(self, input_values: List[OHLCV] = None):
+        super().__init__()
+
+        self.cumsumPV = 0
+        self.cumsumV = 0
+
+        self.initialize(input_values)
+
+    def _calculate_new_value(self) -> Any:
+        value = self.input_values[-1]
+
+        self.cumsumPV += value.volume*(value.high+value.low+value.close)/3
+        self.cumsumV += value.volume
+
+        if (self.cumsumV != 0):
+            return self.cumsumPV/self.cumsumV
+        else:
+            return None
+
+    def remove_input_value(self) -> None:
```

Comment:
If you want to maintain auxiliary variables such as `self.cumsumPV` and `self.cumsumV`, then the correct way to treat them is:

1/ make them lists and register them via `self.add_managed_sequence(...)`. This way you just append a newly calculated value into the list and does not have to bother defining `remove_input_value` at all since the managed lists are handled by the framework. There are many examples in the existing indicators.

2/ If you do not want to use managed sequences, then implement `_remove_input_value_custom()` and implement "undo" step just for the auxiliary variables, do not redefine whole content of `remove_input_value`.

Reply:
Ohh, okay, got it. Thanks for the suggestions. Will try to update it.